### DISCUSSION TITLE
`npm ci` is hanging

### DISCUSSION THREAD

**Discussion by plam1999** [mozilla/hubs#3888] [General Questions & Discussion] [answered]
**Description**
I'm trying to deploy the hubs on Raspberry Pi 3.
But following the guide, `npm ci` is hanging (freezing) at some modules. Mostly, it is hanging at **emojione** module.
I have tried it on Mac OS, It was also hanging at **emoijone** module for git version, but working fine with downloaded zip version.
For Windows OS, I got a really weird errors that I don't have the correct access rights to perform `npm ci` (code 128) for both git and download zip versions.

**To Reproduce**
1. git clone https://github.com/mozilla/hubs.git
2. cd hubs
3. npm ci
4. Hanging

**Screenshots**
Raspberry Pi errors screen shots:
![image](https://user-images.githubusercontent.com/36938625/107332184-60e6b500-6afb-11eb-8200-c73924affe60.png)
![image](https://user-images.githubusercontent.com/36938625/107332216-6c39e080-6afb-11eb-8f02-cd38d2617c70.png)

Windows errors screen shots:
![image](https://user-images.githubusercontent.com/36938625/107332234-74921b80-6afb-11eb-8b55-0bfc96de1eab.png)
![image](https://user-images.githubusercontent.com/36938625/107332269-7cea5680-6afb-11eb-932b-5aa0a6c4ce3d.png)

**Hardware** 
 - Device: Raspberry Pi
 - OS: Raspbian(Debian)

**Comment by robinkwilson** [comment, marked as answer]
Windows command prompt needs to be opened as admin to run `npm ci` successfully. Right click on command prompt start icon to run as admin. (there's a few discussion threads on this topic if you search. Some about the `node` and `npm` versions too).

I don't believe a rasberry pi 3 will successfully compile hubs. This is a little out of scope of what we can support. You're welcome to try though!

Thanks for posting with screenshots!

**Reply by plam1999** [reply]
Thanks Robin,

I got the reason why the pi was hanging now. It is because of the limited network flow when sharing the internet to the pi via LAN cable. The problem should be fixed once it connects directly to the internet router. But unfortunately, Pi 3 B+ model might be not strong enough to host the hubs locally since the pi was not respoding when enter the hubs website via local host. I will give it another go on other stronger Pi.

Thanks for your solution btw.
